functions/src/lobby.ts

```diff
@@ -180,8 +188,64 @@ export const collectPromptAnswers = functions.firestore
 
       transaction.update(lobbyDocRef, { state: "CHAT" });
 
-      functions.logger.log(promptAnswers);
-
-      // TODO #32 create a one on one chat and store the promptAnswers in it
+      // TODO: check if we have a stalker
+      const { pairs } = generatePairs(lobbyData);
+      // create a chatroom for each pair
+      pairs.forEach(async ({ one, two }) => {
+        const room = await getChatRoomCollection(lobbyDocRef).add({ pair: [one, two], viewers: [] });
+        // get answers of the pair
+        const oneAnswer = promptAnswers.get(one) as string;
+        const twoAnswer = promptAnswers.get(two) as string;
+        // place answers in chatMessages inside their room
+        await getChatRoomMessagesCollection(room).add({
+          sender: one,
+          text: oneAnswer,
+          timestamp: firestore.Timestamp.now(),
+          isPromptAnswer: true,
+        });
+        await getChatRoomMessagesCollection(room).add({
+          sender: two,
+          text: twoAnswer,
+          timestamp: firestore.Timestamp.now(),
+          isPromptAnswer: true,
+        });
+      });
     });
   });
+
+export const verifyExpiration = functions.https.onCall(async (data, context) => {
+  // check auth
+  if (context.auth == undefined) {
+    throw new functions.https.HttpsError("permission-denied", "User is not Authenticated");
+  }
+  // check the data request
+  if (!isLobbyRequest(data)) {
+    throw new functions.https.HttpsError("invalid-argument", "Data is not of ExpirationRequest type.");
+  }
+  // get lobby doc
+  const lobbyDocRef = lobbyCollection.doc(data.code);
+
+  // start transaction
+  return db.runTransaction(async (transaction) => {
+    const lobbyDoc = await transaction.get(lobbyDocRef);
+    const lobby = lobbyDoc.data();
+    if (lobby === undefined) {
+      throw new functions.https.HttpsError("not-found", "Lobby does not exist.");
+    }
+    // if the time sent is less than expiration
+    if (Date.now() < lobby.expiration.toMillis()) {
+      throw new functions.https.HttpsError("invalid-argument", "Too early to make request.");
+    }
+    // if the timer sent is equal or greater than
+    // change to the next phase
+
+    // TODO: potential if checks for other states that require a timer
+    // if the state is chat then delete chatrooms
+    if (lobby.state === "CHAT") {
+      await deleteChatRooms(lobby, lobbyDocRef, transaction);
+      return transaction.update(lobbyDocRef, { state: "VOTE" });
```

Comment:
I kind of don't like how this is modified here and inside of `deleteChatRooms` but I guess it's ok